ui-core/src/main/java/com/adyen/checkout/components/ui/view/AdyenComponentView.kt

```diff
@@ -104,7 +106,12 @@ class AdyenComponentView @JvmOverloads constructor(
 
         val localizedContext = context.createLocalizedContext(componentParams.shopperLocale)
 
-        if (viewType is ButtonComponentViewType) {
+        if (viewType is AmountButtonComponentViewType && !componentParams.amount.isEmpty) {
+            binding.payButton.text = localizedContext.getString(
```

Comment:
Can we already add the logic here for `Pay amount` - `Pay` - `Confirm pre-authorization`?